samples/nrf9160/modem_shell/CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.13.1)
+
+#configuration for secure side
+if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/spm.conf")
+  set(spm_CONF_FILE
+    prj.conf
+    ${CMAKE_CURRENT_LIST_DIR}/spm.conf
+  )
+endif()
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(MOSH)
+target_include_directories(app PRIVATE
+  ${CMAKE_CURRENT_SOURCE_DIR}/src
+  ${CMAKE_CURRENT_SOURCE_DIR}/iperf
+  ${CMAKE_CURRENT_SOURCE_DIR}/ping
+  ${CMAKE_CURRENT_SOURCE_DIR}/utils)
+
+if (DEFINED ENV{BUILD_ID})
+  zephyr_compile_definitions(
+    BUILD_ID=$ENV{BUILD_ID}
+  )
+endif()
+
+if (DEFINED ENV{BUILD_VARIANT})
+  zephyr_compile_definitions(
+    BUILD_VARIANT=$ENV{BUILD_VARIANT}
```

Comment:
What is a BUILD_VARIANT, what is it used for, and why is it important ?

Why does this need to be specified as an environment variable, and not through Kconfig or a regualer CMake variable, like `-DBUILD_VARIANT`.
I don't find any documentation on this.

Reply:
See comments on `BUILD_ID`